lib/jsdom/living/xhr/xhr-utils.js

```diff
@@ -297,30 +292,52 @@ function createClient(xhr) {
                   body !== "" &&
                   !(ucMethod === "HEAD" || ucMethod === "GET");
 
-  if (hasBody && !flag.formData) {
-    options.body = body;
-  }
-
   if (hasBody && getRequestHeader(flag.requestHeaders, "content-type") === null) {
     requestHeaders["Content-Type"] = "text/plain;charset=UTF-8";
   }
 
   function doRequest() {
     try {
-      const client = request(options);
-
-      if (hasBody && flag.formData) {
-        const form = client.form();
-        for (const entry of body) {
-          form.append(entry.name, entry.value, entry.options);
+      let requestBody = body;
+      let len = 0;
+      if (hasBody) {
+        if (flag.formData) {
+          requestBody = new FormData();
```

Comment:
It would be nice not to depend on an external library for this, especially since the part of the library that we use is so small. Let's add something like

```js
// TODO: implement https://html.spec.whatwg.org/#multipart-form-data directly instead of using an external library
```